### ISSUE TITLE
Out of Memory Error

### ISSUE THREAD

**Issue by coredev-uk** [SpacingBat3/ReForged#16]
Hello again,

When building the AppImage I get an out of memory error. I recently bumped all my packages bumping to 4.0.3 and forge 7.6.0, this also came with bumps to a few other packages such as electron, vite and node-gyp. I have tested with 4.0.2 and strangely enough the issue is still occurring. I am not sure if this is a Forge issue or an issue with your package compatibility.

I do run multiple makers in my Forge config, I have attempted to remove them and leave the AppImage one and it still occurs. Here is the log:

```
@reforged/maker-appimage 670679: Merging AppImage data and runtime into single file.

<--- Last few GCs --->

[670679:0x6c40000]    34759 ms: Scavenge 1914.6 (1941.5) -> 1898.9 (1941.5) MB, pooled: 630 MB, 0.14 / 0.00 ms  (average mu = 0.951, current mu = 0.927) allocation failure; 
[670679:0x6c40000]    34766 ms: Scavenge 1914.6 (1941.5) -> 1898.9 (1941.5) MB, pooled: 630 MB, 0.15 / 0.00 ms  (average mu = 0.951, current mu = 0.927) allocation failure; 


<--- JS stacktrace --->

FATAL ERROR: invalid array length Allocation failed - JavaScript heap out of memory
----- Native stack trace -----

 1: 0xe17ec6 node::OOMErrorHandler(char const*, v8::OOMDetails const&) [/home/ci/setup-pnpm/node_modules/.bin/nodejs/22.9.0/bin/node]
 2: 0x123b8c0 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/home/ci/setup-pnpm/node_modules/.bin/nodejs/22.9.0/bin/node]
 3: 0x123bb97 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/home/ci/setup-pnpm/node_modules/.bin/nodejs/22.9.0/bin/node]
 4: 0x14696c5  [/home/ci/setup-pnpm/node_modules/.bin/nodejs/22.9.0/bin/node]
 5: 0x189033c  [/home/ci/setup-pnpm/node_modules/.bin/nodejs/22.9.0/bin/node]
 6: 0x7961b7cac476 ```

**Comment by SpacingBat3** [comment]
Huh, it feels like the last operation is the troublesome one, probably a reason on this is that I do most operations in-memory to achieve better speeds. I might check if it is possible to free the buffers that I might not use, especially during buffer merge process.

I should also say I've rewritten how I merge buffers to make it all working within EcmaScript standard types and buffer manipulations, which could be somewhat an issue due to how many conversions I do between objects  and types (I dunno how GC reacts and keeps with it). But if this also occurs before the last patch, I suppose the data itself kept in memory might be just too much for a larger applications and computers with low random-access memory space.

**Comment by SpacingBat3** [comment]
Anyway, this sure is optimization issue. I'll try to check the places where there are memory allocations, objects and other data that might be freed and see if that helps. Getting more info from the engine (snapshots, DevTools info about memory during exec etc.) would also help to troubleshoot it.

**Comment by SpacingBat3** [comment]
Actually, when I put more attention into the logs:
```
FATAL ERROR: invalid array length Allocation failed - JavaScript heap out of memory
```
This just tells me enough what's going on with `4.0.3`, I'm not sure what's about `4.0.2` but I doubt the cause would be the same…

**Comment by coredev-uk** [comment]
Thanks for looking into this. I have not had much of a chance to test it further myself, I am not sure why 4.0.2 did not work, I suspect it might of just cached the package or something.

**Comment by SpacingBat3** [comment]
Also tests pass because the mock app data is rather small in comparison to Electron's usual data size (should say, mock app is not an Electron app). I suppose I could consider generating random binary data within the mock app data in order to test it in more accurate runtime conditions while avoiding compression to affect it for most cases.
Anyway, fix should reach `4.0.4` release… unless I skip this release number due to *unlucky* meaning in CS of it 😅 (bad pun intended).

**Comment by coredev-uk** [comment]
Thank you!
